### DISCUSSION TITLE
AIS-catcher web page, save different views

### DISCUSSION THREAD

**Discussion by g7ruh** [jvde-github/AIS-catcher#178] [Ideas]
Jasper,

Is it possible to be able to store several different web page links, each covering a different area within the range of the dongle? For example, I have two views: the coverage area for my dongle and a zoomed in one of Southampton Docks.

Opening a new web page and zooming in is slow if labels are displayed, or a lot of work to turn labels off, zoom as required and enabling labels again.

Being able to store, say five, different views within the coverage area would allow quick switching in case something interesting occurs on the marine band radio, for example a sailing vessel race start.

thanks for considering this,

Roger

![AIS-catcher-main-coverage-area](https://github.com/jvde-github/AIS-catcher/assets/5949609/e01d9721-56ec-4cac-a168-61b36577c520)

![AIS-catcher-zoomed-in-view](https://github.com/jvde-github/AIS-catcher/assets/5949609/9dcae43c-b82b-4102-ba48-410f84c956e2)

**Comment by jvde-github** [comment]
Yes, you can give all settings (active tab, location, zoom, etc) as URL parameters and this you can bookmark in the browser. Will show you example when back at my PC.

**Comment by jvde-github** [comment]
Added a few lines to the program. You can select: 
![image](https://github.com/jvde-github/AIS-catcher/assets/52420030/ced4039b-7238-4652-bbbe-ef3dae177ae1)

Which will then copy a URL to the clipboard that opens the map at the current location and zoom level. You can bookmark this link. 

The link will be something like this:
```
http://raspberrypi:8100?lat=17.55626&lon=7.24571&zoom=7.3500000000000005&tab=map
```
Not heavily tested for now....

**Reply by g7ruh** [reply]
Jasper,

Thanks, works as requested in Firefox. I created four different views and changing them quite often, it works fine.

Will test some more over the next few days.

Roger

**Reply by g7ruh** [reply]
Jasper,

Tested in Edge and in Portable Apps Chrome: works as in Firefox

Roger

**Comment by Taxom** [comment]
Hello Jasper!
Is it possible to make this link to immediately display this link in the address bar, and changed when the map is changed scale and the place? How to happen in google maps.
And another question, is it possible to make a click in the scroll button on the menu buttons open them in a new tab?
Sometimes I need to open the list in the new tab parallel with the map and I have to separately open the tab and open the second page on which to switch to the list, which is a little inconvenient. Thank you!
Max

**Comment by jvde-github** [comment]
Good suggestion. I have removed the menu item and now update the address bar. Saves cluttering the menus. 

I don't understand the comment about Google Maps.

You want to open the ships and map tab at the same time? Why not create a bookmark for both tabs (?tab=ships and ?tab=map) and open both bookmarks at the same time?

**Reply by Taxom** [reply]
Google Maps was just for example this is how the the address line looks like 
![maps exaple](https://github.com/jvde-github/AIS-catcher/assets/20313019/dad78dda-fe66-4f48-83ba-773e22b3e444)

Of course, I can make a few bookmarks, but it’s easier to open it in a new tab with one click on the button, but this is rather a matter of habit. I’m just used to opening the links in the new tab in this way, it seems more convenient to me. So if it takes time and coding, then it is not necessary at all.

**Reply by jvde-github** [reply]
Ok, clear. I think the URL concept is now implemented as your Google Maps example. 

I will think about the other idea. I plan all along was to implement a ship overview table on the map tab (bit like tar1090). Maybe that is a better way.

**Comment by Taxom** [comment]
Yes table like in tar1090 with search, sort, and filter would be great!

**Reply by jvde-github** [reply]
I have thought about it and will add it in the same UI element as the summary window (ie. opens left of the controls). Will be more limited though than the main table as I find that tabulator (the library I am using for that table) has a bit too much overhead. Also will add a Filter option to this.

**Reply by jvde-github** [reply]
First version built-in. Still needs some tidying up and a second button for search, filter and sort.

![Screenshot from 2023-10-07 07-31-30](https://github.com/jvde-github/AIS-catcher/assets/52420030/f0b79dc3-54e9-475b-81a6-8c9fb18ad909)

**Reply by Taxom** [reply]
I have not had time to update yet, but on the KX1T website does not work the scrolling in this table. Instead the scale of the map is changing.

**Reply by jvde-github** [reply]
Yes, only works with the scrollbar on the right (at least on Firefox and Chrome)

**Reply by jvde-github** [reply]
I have made some changes and you can also now use the scrollwheel if mouse pointer is on top of the ship table. Having said that, still some work to be done to complete table.

**Reply by jvde-github** [reply]
Ok, sorting is implemented in this new table and moved table to the right of the map controls. Feels more intuitive. Need to start working on a simple search and filter function.
